### PR TITLE
Add mutable field

### PR DESCRIPTION
**Pull request by sporejack**
TBA

### CODE REVIEW

File: src/book.rs [hunk 1]
```diff
@@ -233,8 +234,10 @@ impl Book {
                 )
                 .await;
 
-                if opposite.amount.is_zero() { /* TODO: delete ask order */ }
+                if opposite.amount_left.is_zero() { /* TODO: delete ask order */
+                }
 
+                dbg!(&running_total, &amount);
```
Comment (raymogg) [inline src/book.rs]: Hanging dbg

**Review by raymogg — CHANGES_REQUESTED** [review]
Just confirming (as it doesn't appear to be in the changes), the value getting passed to the executioner is still the original amount right, and not amount_left?

Otherwise lgtm

File: src/book.rs (hunk 1 above)
Comment (sporejack) [reply src/book.rs]: Consult 7b21f18.

**Comment by sporejack** [comment]
> Just confirming (as it doesn't appear to be in the changes), the value getting passed to the executioner is still the original amount right, and not amount_left?

So, the way I've approached this is to instruct `serde` to ignore the `amount_left` field whenever its serialising or deserialising. This is achieved via a macro here:

https://github.com/tracer-protocol/tracer-ome/blob/7b21f181ccf5d9cc2f3d0da08d3a7a4822fdc530/src/order.rs#L64

**Review by raymogg — APPROVED** [review]

**Pull request merged**